src/commands/copy/copy-rpc.ts

```diff
@@ -17,63 +17,83 @@ export interface CopyContractArgs {
   /** Number of records to copy */
   size: number;
 
-  /** Force overwrite files.
-   * When `deleteSource` is set, this will delete source files after a forced copy. */
+  /**
+   * Force overwrite files.
+   * When `deleteSource` is set, this will delete source files after a forced copy.
+   */
   force: boolean;
 
-  /** Do not overwrite existing files but skip them if they are the same size.
-   * When `deleteSource` is set, this will delete source files that get skipped. */
+  /**
+   * Do not overwrite existing files but skip them if they are the same size.
+   * When `deleteSource` is set, this will delete source files that get skipped.
+   */
   noClobber: boolean;
 
   /** Correct content-type from "application/octet-stream" to common formats */
   fixContentType: boolean;
 
-  /** Compress files using zstandard while copying.
+  /**
+   * Compress files using zstandard while copying.
    * Will copy the uncompressed file if the compressed output is likely to be larger than the input (e.g. for very small files <90 bytes).
-   * When compressing data, this will append `.zst` to the target file name. */
+   * When compressing data, this will append `.zst` to the target file name.
+   */
   compress: boolean;
 
+  /**
+   * Decompress .zst files while copying.
+   * Will copy the file if the file name does not end in `.zst`.
+   * When decompressing data, this will remove `.zst` from the target file name.
+   */
+  decompress: boolean;
+
   /** Delete source files after copying or compressing */
   deleteSource: boolean;
 }
 
 export interface CopyStats {
   /** Number of files copied */
-  copied: number;
-  /** Number of bytes copied */
-  copiedBytes: number;
+  copied: { count: number; bytes: number };
+
   /** Number of files compressed */
-  compressed: number;
-  /** Number of bytes compressed (input) */
-  compressedInputBytes: number;
-  /** Number of bytes compressed (output) */
-  compressedOutputBytes: number;
+  compressed: { count: number; bytesIn: number; bytesOut: number };
+
+  /** Number of files decompressed */
+  decompressed: { count: number; bytesIn: number; bytesOut: number };
+
   /** Number of source files deleted */
-  deleted: number;
-  /** Number of bytes deleted */
-  deletedBytes: number;
-  /** Number of times a file was retried */
-  retries: number;
-  /** Number of files that have been skipped, generally because the target file already exists */
-  skipped: number;
-  /** Number of bytes that has been skipped. */
-  skippedBytes: number;
+  deleted: { count: number; bytes: number };
```

Comment:
would it be easier to just have this as `bytesIn` and `bytesOut` empty so the interface is the same across all stats?

then you can do something like 
```typescript
joinStats(a: stats, b: stats):stats { 
  return { count: a.count + b.count, bytesIn: a.bytesIn + b.bytesIn, bytesOut: a.bytesOut b.bytesOut }
}